### ISSUE TITLE
Fix matching of source_id when observation_id includes station_id with hyphen in it

### ISSUE THREAD

**Issue by agstephens** [glamod/glamod-ingest#48]
My code splits on the hyphen, so you are right, I might need to rerun everything. 

I currently do this:

`primary_id, record_number = x['observation_id'].split('-')[:2]`

So, I would convert this as follows:

```
id = " BR00B6-0360-1-1965-01-01-44-13"
primary_id, record_number = id.split('-')[:2]
primary_id == 'BR00B6'
record_number == '0360'
```

What is the foolproof method of breaking it up?

- Does it always end with: <YYYY>-<MM>-<DD>-<observed_variable>-<value_significance> ? 
- If so, I can split off the last 5 items first…
- …then:
  - record_number = last item
  - primary_id = first items joined by '-'

Will that do the job?

**Comment by agstephens** [comment]
A new method of working back on the split would work, the files  will always have :

```
Daily:  <YYYY>-<MM>-<DD>-<observed_variable>-<value_significance>
Sub daily : <YYYY>-<MM>-<DD>-<HH:MM>-<observed_variable>-<value_significance>
MONTHLY : <YYYY>-<MM>-<observed_variable>-<value_significance>
```

**Comment by agstephens** [comment]
Implemented a fix in: b7c21d1dc5d6d8ece7ed0e6a68de0fcb0abb9df0
